src/VeeArc.Infrastructure/DataBase/Interceptors/AuditableEntitySaveChangesInterceptor.cs

```diff
@@ -0,0 +1,41 @@
+using System.Runtime.InteropServices.JavaScript;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Diagnostics;
+using VeeArc.Domain.Common;
+
+namespace VeeArc.Infrastructure.Persistence.Interceptors;
+
+public class AuditableEntitySaveChangesInterceptor : SaveChangesInterceptor
+{
+    public override InterceptionResult<int> SavingChanges(DbContextEventData eventData, InterceptionResult<int> result)
+    {
+        UpdateEntities(eventData.Context);
+
+        return base.SavingChanges(eventData, result);
+    }
+
+    public override ValueTask<InterceptionResult<int>> SavingChangesAsync(DbContextEventData eventData, InterceptionResult<int> result, CancellationToken cancellationToken = default)
+    {
+        UpdateEntities(eventData.Context);
+
+        return base.SavingChangesAsync(eventData, result, cancellationToken);
+    }
+
+    public void UpdateEntities(DbContext? context)
+    {
+        if (context == null) return;
+
+        foreach (var entry in context.ChangeTracker.Entries<BaseAuditableEntity>())
+        {
+            if (entry.State is EntityState.Added)
```

Comment:
Maybe make this if more readable? Create private methods for this

Reply:
yobnuv switch

Reply:
switch is not cool)